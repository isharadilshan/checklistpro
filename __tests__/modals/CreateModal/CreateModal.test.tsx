import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import CreateModal from '../../../src/components/modals/CreateModal';

configure({adapter: new Adapter()});

const props = {
  isModalVisible: true,
  closeModal: jest.fn(),
  onCreate: jest.fn(),
};

describe('when rendering', () => {
  const wrapper = shallow(<CreateModal {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
