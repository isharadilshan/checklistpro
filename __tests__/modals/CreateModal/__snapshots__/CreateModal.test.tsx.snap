// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<ModalWrapper
  closeModal={[MockFunction]}
  modalVisible={true}
>
  <View
    style={
      {
        "padding": 16,
      }
    }
  >
    <Component
      control={
        {
          "_defaultValues": {},
          "_executeSchema": [Function],
          "_fields": {},
          "_formState": {
            "dirtyFields": {},
            "errors": {},
            "isDirty": false,
            "isSubmitSuccessful": false,
            "isSubmitted": false,
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "submitCount": 0,
            "touchedFields": {},
          },
          "_formValues": {},
          "_getDirty": [Function],
          "_getFieldArray": [Function],
          "_getWatch": [Function],
          "_names": {
            "array": Set {},
            "mount": Set {},
            "unMount": Set {},
            "watch": Set {},
          },
          "_options": {
            "mode": "onSubmit",
            "reValidateMode": "onChange",
            "shouldFocusError": true,
          },
          "_proxyFormState": {
            "dirtyFields": false,
            "errors": false,
            "isDirty": false,
            "isValid": false,
            "isValidating": false,
            "touchedFields": false,
          },
          "_removeUnmounted": [Function],
          "_stateFlags": {
            "action": false,
            "mount": false,
            "watch": false,
          },
          "_subjects": {
            "array": {
              "next": [Function],
              "observers": [],
              "subscribe": [Function],
              "unsubscribe": [Function],
            },
            "state": {
              "next": [Function],
              "observers": [],
              "subscribe": [Function],
              "unsubscribe": [Function],
            },
            "watch": {
              "next": [Function],
              "observers": [],
              "subscribe": [Function],
              "unsubscribe": [Function],
            },
          },
          "_updateFieldArray": [Function],
          "_updateValid": [Function],
          "getFieldState": [Function],
          "register": [Function],
          "unregister": [Function],
        }
      }
      defaultValue=""
      name="title"
      render={[Function]}
      rules={
        {
          "required": true,
        }
      }
    />
    <Component
      control={
        {
          "_defaultValues": {},
          "_executeSchema": [Function],
          "_fields": {},
          "_formState": {
            "dirtyFields": {},
            "errors": {},
            "isDirty": false,
            "isSubmitSuccessful": false,
            "isSubmitted": false,
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "submitCount": 0,
            "touchedFields": {},
          },
          "_formValues": {},
          "_getDirty": [Function],
          "_getFieldArray": [Function],
          "_getWatch": [Function],
          "_names": {
            "array": Set {},
            "mount": Set {},
            "unMount": Set {},
            "watch": Set {},
          },
          "_options": {
            "mode": "onSubmit",
            "reValidateMode": "onChange",
            "shouldFocusError": true,
          },
          "_proxyFormState": {
            "dirtyFields": false,
            "errors": false,
            "isDirty": false,
            "isValid": false,
            "isValidating": false,
            "touchedFields": false,
          },
          "_removeUnmounted": [Function],
          "_stateFlags": {
            "action": false,
            "mount": false,
            "watch": false,
          },
          "_subjects": {
            "array": {
              "next": [Function],
              "observers": [],
              "subscribe": [Function],
              "unsubscribe": [Function],
            },
            "state": {
              "next": [Function],
              "observers": [],
              "subscribe": [Function],
              "unsubscribe": [Function],
            },
            "watch": {
              "next": [Function],
              "observers": [],
              "subscribe": [Function],
              "unsubscribe": [Function],
            },
          },
          "_updateFieldArray": [Function],
          "_updateValid": [Function],
          "getFieldState": [Function],
          "register": [Function],
          "unregister": [Function],
        }
      }
      defaultValue=""
      name="description"
      render={[Function]}
      rules={
        {
          "required": true,
        }
      }
    />
    <Button
      label="Create"
      mode="contained"
      onPress={[Function]}
      style={
        {
          "marginHorizontal": 40,
          "marginTop": 45,
        }
      }
    />
  </View>
</ModalWrapper>
`;
